service/src/main/java/com/codeforcommunity/processor/TeamsProcessorImpl.java

```diff
@@ -35,4 +74,279 @@ public void disbandTeam(JWTData userData, int teamId) {
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.USER_ID.eq(memberId)))
+              .fetchOne();
+      if (approvedUserRecord.getTeamRole() == TeamRole.PENDING) {
+        approvedUserRecord.setTeamRole(TeamRole.MEMBER);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(
+            memberId, teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          leaderTeamsRecord.getTeamId(), leaderTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void rejectUser(JWTData userData, int teamId, int memberId) {
+    checkUserExists(memberId);
+    checkTeamExists(teamId);
+    UsersTeamsRecord leaderTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+    if (leaderTeamsRecord != null && leaderTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      UsersTeamsRecord approvedUserRecord =
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.TEAM_ID.eq(teamId).and(USERS_TEAMS.USER_ID.eq(memberId))))
+              .fetchOne();
+      if (approvedUserRecord.getTeamRole() == TeamRole.PENDING) {
+        approvedUserRecord.setTeamRole(TeamRole.None);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(
+            userData.getUserId(), teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          leaderTeamsRecord.getTeamId(), leaderTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void kickUser(JWTData userData, int teamId, int memberId) {
+    checkUserExists(memberId);
+    checkTeamExists(teamId);
+    UsersTeamsRecord leaderTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+    if (leaderTeamsRecord != null && leaderTeamsRecord.getTeamRole() == TeamRole.LEADER) {
+      UsersTeamsRecord approvedUserRecord =
+          db.selectFrom(USERS_TEAMS.where(USERS_TEAMS.USER_ID.eq(memberId)))
+              .fetchOne();
+      if (approvedUserRecord.getTeamRole() == TeamRole.MEMBER) {
+        approvedUserRecord.setTeamRole(TeamRole.None);
+        approvedUserRecord.update();
+      } else {
+        throw new MemberStatusException(memberId, teamId);
+      }
+    } else {
+      throw new WrongTeamRoleException(
+          leaderTeamsRecord.getTeamId(), leaderTeamsRecord.getTeamRole());
+    }
+  }
+
+  @Override
+  public void leaveTeam(JWTData userData, int teamId) {
+    checkTeamExists(teamId);
+    UsersTeamsRecord usersTeamsRecord =
+        db.selectFrom(USERS_TEAMS)
+            .where(USERS_TEAMS.USER_ID.eq(userData.getUserId()))
+            .and(USERS_TEAMS.TEAM_ID.eq(teamId))
+            .fetchOne();
+    if (usersTeamsRecord != null && usersTeamsRecord.getTeamRole() != TeamRole.LEADER) {
```

Comment:
This is probably a case where we would want to have a message if a leader tries to leave their own team that's different from the message you get if you're not on the team at all. You can create an improvement ticket for this

Reply:
I created a different Exception for this